manifests/init.pp

```diff
@@ -238,192 +185,101 @@
+    apache_http_vhost         => $apache_http_vhost,
+    apache_https_vhost        => $apache_https_vhost,
+    apache_https_cert         => $apache_https_cert,
+    apache_https_key          => $apache_https_key,
+    apache_https_ca           => $apache_https_ca,
+    apache_https_chain        => $apache_https_chain,
+    apache_vhost_priority     => $priority,
+    servername                => $servername,
+    static_url                => '/pulp/assets/',
+    postgresql_manage_db      => $pulpcore_manage_postgresql,
+    postgresql_db_host        => $pulpcore_postgresql_host,
+    postgresql_db_port        => $pulpcore_postgresql_port,
+    postgresql_db_user        => $pulpcore_postgresql_user,
+    postgresql_db_password    => $pulpcore_postgresql_password,
+    postgresql_db_name        => $pulpcore_postgresql_db_name,
+    postgresql_db_ssl         => $pulpcore_postgresql_ssl,
+    postgresql_db_ssl_require => $pulpcore_postgresql_ssl_require,
+    postgresql_db_ssl_cert    => $pulpcore_postgresql_ssl_cert,
+    postgresql_db_ssl_key     => $pulpcore_postgresql_ssl_key,
+    postgresql_db_ssl_root_ca => $pulpcore_postgresql_ssl_root_ca,
+    worker_count              => $pulpcore_worker_count,
+    before                    => Class['foreman_proxy::plugin::pulp'],
   }
 
-  if $pulpcore {
-    if $pulpcore_mirror {
-      $pulpcore_allowed_import_path    = ['/var/lib/pulp/sync_imports']
-      $pulpcore_allowed_export_path    = []
-    } else {
-      $pulpcore_allowed_import_path    = ['/var/lib/pulp/sync_imports', '/var/lib/pulp/imports']
-      $pulpcore_allowed_export_path    = ['/var/lib/pulp/exports']
-    }
-    if $shared_with_foreman_vhost {
-      include foreman::config::apache
-      $servername = $foreman::config::apache::servername
-      $priority = $foreman::config::apache::priority
-      $apache_http_vhost = 'foreman'
-      $apache_https_vhost = 'foreman-ssl'
-      $apache_https_cert = undef
-      $apache_https_key = undef
-      $apache_https_ca = undef
-      $apache_https_chain = undef
-      Class['foreman::config::apache'] -> Class['pulpcore::apache']
-    } elsif $pulp and $pulp::manage_httpd {
-      $servername = $pulp::server_name
-      $priority = '05'
-      $apache_http_vhost = 'pulp-http'
-      $apache_https_vhost = 'pulp-https'
-      $apache_https_cert = undef
-      $apache_https_key = undef
-      $apache_https_ca = undef
-      $apache_https_chain = undef
-      Class['pulp::apache'] -> Class['pulpcore::apache']
-    } else {
-      include certs::apache
-      Class['certs::apache'] ~> Class['pulpcore::apache']
-      $servername = $certs::apache::hostname
-      $priority = undef
-      $apache_http_vhost = undef
-      $apache_https_vhost = undef
-      $apache_https_cert = $certs::apache::apache_cert
-      $apache_https_key = $certs::apache::apache_key
-      $apache_https_ca = $certs::katello_default_ca_cert
-      $apache_https_chain = $certs::katello_server_ca_cert
+  if $enable_docker {
```

Comment:
If we're doing a breaking change, thoughts about renaming this to `$enable_container`?

Reply:
I was thinking a similar thing. Given this will require an installer migration alongside it, I thought to do it as an individual follow up change after this all went in.

Reply:
My reasoning was that it already required an installer migration anyway. Might as well combine them into one.